extensions/resource-deployment/src/services/platformService.ts

```diff
@@ -83,12 +90,24 @@ export class PlatformService implements IPlatformService {
 		this._outputChannel.show(preserveFocus);
 	}
 
-	osType(platform: string = this.platform()): OsType {
+	osType(): OsType {
```

Comment:
>osType [](start = 1, length = 6)

It would be preferable to make this an async function instead that awaits on the initialization promise if it isn't completed yet. I'd suggest naming it getOsType and then rename the current one to initOsType

Reply:
getOsType already exists as a private async function. Do you want the public OsType to wait if the PlatformService has been constructed but not yet initialized instead of throwing?

Reply:
Yes. Throwing an error is really not desirable behavior. Having the caller be able to wait until the osType is initialized is a much better experience and will prevent random issues arising from the possibility of the osType initialization not happening immediately.